packages/marshal/src/passStyleOf.js

```diff
@@ -235,50 +286,99 @@ const assertRemotableProto = val => {
  * mustPassByRemote.
  *
  * @param {*} val The remotable candidate to check
+ * @param {Checker} [check]
+ * @returns {boolean}
  */
-export function assertCanBeRemotable(val) {
-  // throws exception if cannot
-  assert.typeof(val, 'object', X`cannot serialize non-objects like ${val}`);
-  assert(!Array.isArray(val), X`Arrays cannot be pass-by-remote`);
-  assert(val !== null, X`null cannot be pass-by-remote`);
+function checkCanBeRemotable(val, check = x => x) {
+  if (
+    !(
+      check(
+        typeof val === 'object',
+        X`cannot serialize non-objects like ${val}`,
+      ) &&
+      check(!Array.isArray(val), X`Arrays cannot be pass-by-remote`) &&
+      check(val !== null, X`null cannot be pass-by-remote`)
+    )
+  ) {
+    return false;
+  }
 
   const descs = getOwnPropertyDescriptors(val);
   const keys = ownKeys(descs); // enumerable-and-not, string-or-Symbol
-  keys.forEach(key => {
-    assert(
+  return keys.every(
+    key =>
       // Typecast needed due to https://github.com/microsoft/TypeScript/issues/1863
-      !('get' in descs[/** @type {string} */ (key)]),
-      X`cannot serialize objects with getters like ${q(String(key))} in ${val}`,
-    );
-    assert.typeof(
-      // @ts-ignore https://github.com/microsoft/TypeScript/issues/1863
-      val[key],
-      'function',
-      X`cannot serialize objects with non-methods like ${q(
-        String(key),
-      )} in ${val}`,
-    );
-    assert(
-      key !== PASS_STYLE,
-      X`A pass-by-remote cannot shadow ${q(PASS_STYLE)}`,
-    );
-  });
+      check(
+        !('get' in descs[/** @type {string} */ (key)]),
```

Comment:
Would it be better to use the `@ts-ignore` pattern you used on line 260?

Reply:
I avoid `@ts-ignore` whenever I can.  It's too easy to break something and not notice.  The cast is not nice, but it's nicer than `@ts-ignore`.